cycles-ledger/src/lib.rs

```diff
@@ -177,6 +178,44 @@ pub fn transfer_from_error_to_transfer_error(e: TransferFromError) -> TransferEr
     }
 }
 
+// Traps if the error is InsufficientAllowance
```

Comment:
```suggestion
// Traps if the error is [InsufficientAllowance]
```

Reply:
Why does it trap in this case?
Why isn't this error handled the same way as `InsufficientFunds`?

Reply:
Because in `withdraw` there is no concept of allowances. If allowances are a problem in `withdraw`, then that's a bug